packages/blade/src/components/BaseHeaderFooter/BaseHeader.tsx

```diff
@@ -172,48 +194,61 @@ const _BaseHeader = ({
             flex="auto"
             display="flex"
             flexDirection="row"
-            alignItems="center"
+            alignItems="flex-start"
           >
             {leading ? (
               <BaseBox
                 width="spacing.8"
                 height="spacing.8"
-                flexShrink={0}
                 marginRight="spacing.3"
-                justifyContent="center"
-                alignItems="center"
-                display="flex"
+                {...centerBoxProps}
               >
                 {leading}
               </BaseBox>
             ) : null}
-            <BaseBox>
-              <BaseBox flexShrink={0} display="flex" flexDirection="row" alignItems="center">
+            <BaseBox flex="auto">
+              <BaseBox
+                // Explicitly setting maxWidth in React Native because text is not being wrapped properly when multiple fix width components are rendered in header
```

Comment:
So this means native will always have a max width of 100px  for title? That seems to a restriction that I'm not sure if it'll be good enough. What if there's a long title and not a lot of other components on header?

Reply:
It also has `shouldWrapTitle` condition on that line which checks if leading, trailing, and other things exists. So it only sets that maxWidth when other elements are also present.
<img width="834" alt="image" src="https://github.com/razorpay/blade/assets/30949385/dd837993-af20-4a76-b704-a5dd6bc8f202">

Ideally it will be rare to have all of those things together in mobile but added a check just in case.